sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,500 @@
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<response_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked). 
+
+| invocation_capacity | The `invocation_capacity` property defines the minimum
+required number `Size` of `async` invocations that must be invokable without any
+`get` being called on that same set.  This number of `async` invocations without
+a `get` call is the minimum number that will be supported before the `async`
+member function blocks.
+
+| response_capacity | The `response_capacity` property defines the maximum
+number `Size` of outstanding `async` invocations such that all outstanding
+invocations are guaranteed to make forward progress. Further `async` invocations
+beyond `Size` may block until enough `get` calls are invoked such that the
+number of outstanding `async` invocations is reduced to `Size`.
+|===
+--
+
+=== Compatibility with FPGA Kernel Interface Properties
+
+The `task_sequence` may be declared with the FPGA Kernel Interface `pipelined`
+property described in link:sycl_ext_oneapi_kernel_properties.asciidoc[
+  sycl_ext_oneapi_kernel_properties].
+Normally the `pipelined` property is applicable only on kernels however this
+extension supports applying the property on task sequences.
+
+== Forward Progress Guarantees and Execution Model
+
+C{plus}{plus} defines a framework for describing the
+http://eel.is/c++draft/intro.progress[forward progress] of individual threads
+with respect to one another in a multi-threaded program. 
+
+Applying this framework to `task_sequence`, `async` tasks belonging to the same
+`task_sequence` object provide a http://eel.is/c++draft/intro.progress#11[
+weakly parallel forward progress guarantee]. That is, tasks belonging to a
+`task_sequence` object are not guaranteed to make forward progress with respect
+to each other. 
+
+Tasks belonging to different `task_sequence` objects provide a
+http://eel.is/c++draft/intro.progress#9[parallel forward progress guarantee]
+with respect to each other. That is, tasks belonging to different
+`task_sequence` objects are guaranteed to make forward progress with respect to
+each other once they have been initiated.
+
+Tasks belonging to a `task_sequence` object provide a
+http://eel.is/c++draft/intro.progress#9[parallel forward progress guarantee]
+with respect to their caller. That is, a `task_sequence`'s tasks are guaranteed
+to make forward progress with respect to their caller (that invoked `async`).
+
```

Comment:
Maybe add to this section what I mentioned above about how threads block with forward progress guarantee delegation so long as the number of outstanding asyncs is less than response capacity.

Reply:
Added.